llvm/lib/Target/AMDGPU/AMDGPURegBankLegalizeHelper.cpp

```diff
@@ -237,6 +237,21 @@ void RegBankLegalizeHelper::lowerS_BFE(MachineInstr &MI) {
   MI.eraseFromParent();
 }
 
+void RegBankLegalizeHelper::lowerSplitTo32(MachineInstr &MI) {
+  Register Dst = MI.getOperand(0).getReg();
+  LLT Ty = MRI.getType(Dst) == V4S16 ? V2S16 : S32;
```

Comment:
```suggestion
  LLT Ty = (MRI.getType(Dst) == V4S16 ? V2S16 : S32);
```

nit for clarity; are you also expecting the type of be something else specifically if it isn't `V4S16` ? If so I'd add an assert to avoid silent failures